code/modules/atmospherics/machinery/atmospherics.dm

```diff
@@ -163,11 +169,14 @@ Pipelines + Other Objects -> Pipe network
 
 //(De)construction
 /obj/machinery/atmospherics/attackby(obj/item/W, mob/user)
+	var/turf/T = get_turf(src)
 	if(can_unwrench && istype(W, /obj/item/wrench))
-		var/turf/T = get_turf(src)
+		if(T.transparent_floor)
+			to_chat(user, "<span class='danger'>You can't interact with something that's under the floor!</span>")
+			return
```

Comment:
I don't like the idea of having these `if(T.transparent_floor)` checks all over, even on machines that are above floors... but I can't really think of a good way to do it otherwise without getting extremely complicated with images or something.